### PR TITLE
fix: filter out module info classes and class node with null superName, closes MISC-365

### PR DESCRIPTION
**Pull request by tpetillot**
**Motivation:**
Module info are not used and have null superName while not being Object,
causing NPE.

**Modification:**
* Filter module info classes and class node with null superName
* Add module-info.class in test jar

### CODE REVIEW

**Review by slandelle — APPROVED** [review]

File: src/main/java/io/gatling/scanner/AsmSimulationScanner.java [hunk 1]
```diff
@@ -81,7 +82,11 @@ private static boolean checkJavaClassMagicBytes(byte[] bytes) {
       throws IOException {
     return bytesFromJarEntry(jar, entry)
         .flatMap(AsmSimulationScanner::classNodeFromBytes)
-        .filter(classNode -> !classNode.superName.equals(ROOT_CLASS_NAME))
+        .filter(
+            classNode ->
+                !classNode.name.equals(MODULE_INFO_NAME)
+                    && classNode.superName != null
+                    && !classNode.superName.equals(ROOT_CLASS_NAME))
```
Comment (Isammoc) [inline src/main/java/io/gatling/scanner/AsmSimulationScanner.java:89]: Yoda style:
```suggestion
                    && !ROOT_CLASS_NAME.equals(classNode.superName))
```

File: src/main/java/io/gatling/scanner/AsmSimulationScanner.java [hunk 2]
```diff
@@ -81,7 +82,11 @@ private static boolean checkJavaClassMagicBytes(byte[] bytes) {
       throws IOException {
     return bytesFromJarEntry(jar, entry)
         .flatMap(AsmSimulationScanner::classNodeFromBytes)
```
[the remark this answers is not in the document]
Comment (slandelle) [reply src/main/java/io/gatling/scanner/AsmSimulationScanner.java:84]: No. It returns an `Optional`.

File: src/main/java/io/gatling/scanner/AsmSimulationScanner.java (hunk 1 above)
Comment (slandelle) [reply src/main/java/io/gatling/scanner/AsmSimulationScanner.java:89]: Well, classNode.superName can't be null thanks to the guard line 88.

Comment (Isammoc) [reply src/main/java/io/gatling/scanner/AsmSimulationScanner.java:89]: Indeed, but in java land, we should follow java best practice.

**Pull request merged**